aiden-recommender/aiden_recommender/scrapers/scraper_aggregator.py

```diff
@@ -0,0 +1,50 @@
+from aiden_recommender.scrapers.wtj.scraper import WelcomeToTheJungleScraper
+from aiden_recommender.scrapers.france_travail.scraper import FranceTravailScraper
+from aiden_recommender import JOB_COLLECTION, qdrant_client
+from aiden_recommender.scrapers.models import JobOffer
+import json
+from mistralai.client import MistralClient
+import os
+from uuid import UUID
+from qdrant_client import models
+
+
+class ScraperAggregator:
+    wtj_scraper = WelcomeToTheJungleScraper()
+    france_travail_scraper = FranceTravailScraper()
+    mistral_client = MistralClient(api_key=os.environ.get("MISTRAL_API_KEY"))
+
+    @classmethod
+    def _get_search_query_vector(cls, search_query: str) -> list[float]:
+        return cls.mistral_client.embeddings(model="mistral-embed", input=[search_query]).data[0].embedding
+
+    @classmethod
+    def search_jobs(cls, search_query: str, location: str, profile_embedding_id: UUID, num_results: int = 15) -> list[str]:
+        user_profile = qdrant_client.rest.points_api.get_point(collection_name="user_profile", id=str(profile_embedding_id)).result
+        if not user_profile:
+            raise RuntimeError("User not present found in vector DB")
+        if (user_vector := user_profile.vector) is None:
+            raise RuntimeError("User has no vector embedding.")
+
+        wtj_embedding_ids = cls.wtj_scraper.search_jobs(search_query=search_query, location=location, num_results=num_results * 10)
+        france_travail_embedding_ids = cls.france_travail_scraper.search_jobs(
+            search_query=search_query, location=location, num_results=num_results * 10
+        )
+
+        # Get embedding ids
+        embedding_ids = wtj_embedding_ids + france_travail_embedding_ids
+
+        # Generate the search vector
+        search_query_vector = cls._get_search_query_vector(search_query + " " + location)
+        search_vector = [a + (b * 0.5) for a, b in zip(search_query_vector, user_vector)]
+
+        # Filter the search to only include the job offers with the given embedding ids
```

Comment:
@Smirkey maybe you now qrandt enough to write a filter to only return elements in the ids just scraped? This would ensure that we respect the demanded location for example.

Reply:
i'll look into it